src/components/menu/menu.tsx

```diff
@@ -80,34 +134,110 @@ export class Menu {
      * Is emitted when a menu item is selected.
      */
     @Event()
-    private select: EventEmitter<MenuItem | MenuItem[]>;
+    public select: EventEmitter<MenuItem | MenuItem[]>;
```

Comment:
I know @jgroth tends to mark these as public, but I don't know why (Johan, want to chime in here?), as the docs include the events even if the event _emitter_ is private, and I don't understand why anyone from outside the component should be allowed to call the event emitter.

Plus, if this is marked `public`, we get no warning if it's unused inside the component. I think it's quite handy to know if it's unused, as we could then remove it. No outside code should call it anyway, right? 🤷‍♂️

Reply:
No one will ever be able to use any of the public properties on the component class, not even the `@Prop`s, since Stencil will just use the class we are implementing and do some magic with it and put it inside the actual web component. My guess is that it will work no matter if we use `public`, `private` or `protected`, since it is just synthetic sugar for TypeScript anyway. If we start making them private for real, using the `#property` syntax, it will probably start to fail.

I like to make them `public` just because they are part of the components public API :smile: In the Stencil docs they don't specify anything, which makes them implicitly `public`

Reply:
Ì agree with @jgroth to have them public.

I like your angle with the unused stuff @adrianschmidt but in some cases you must define the event because a child you render emits an event that you just pass up through your component, and in that case you define it and won't "use" it. And you will get a warning that is incorrect.

Do we agree on having them public?